### PR TITLE
fix(cli): open assistant links in Ghostty

### PR DESCRIPTION
**Pull request by iscekic**
## Summary
Makes HTTP(S) links in Kilo CLI assistant responses clickable in macOS Ghostty, while leaving OpenTUI's native OSC 8 hyperlink handling (and native Cmd-click on macOS / Ctrl-click on Windows/Linux) untouched on every other terminal.

Ghostty reports Cmd-click as an ordinary left click because SGR mouse reporting can't encode the Command modifier there, so a plain left click on an assistant HTTP(S) link is treated as the intended open gesture, scoped strictly to macOS Ghostty and to final assistant-answer text.

## What changed
- `packages/opencode/src/kilocode/cli/cmd/tui/assistant-link.ts` (new): `AssistantLink` namespace — Ghostty detection, assistant-text scoping, native OpenTUI hyperlink lookup, a read-only bare-URL character scan fallback (with word-wrap continuation), scheme validation, and browser launch with retryable-failure reporting.
- `packages/opencode/src/cli/cmd/tui/app.tsx`: wires `AssistantLink.handle()` into the existing root `onMouseUp` handler ahead of copy-on-select, behind an isolated `kilocode_change` block.
- `packages/opencode/src/cli/cmd/tui/routes/session/index.tsx`: gives the final assistant answer text (`TextPart`) a dedicated `assistant-text-` id so reasoning blocks, dialogs, and tool output stay out of scope.
- `packages/opencode/test/kilocode/cli/cmd/tui/assistant-link.test.tsx` (new): real-`@opentui/solid`-renderer tests, not mocks.
- `.changeset/cli-ghostty-assistant-links.md`: patch release note for `@kilocode/cli`.

## A verified scope adjustment from the original plan
While implementing, I rendered real markdown through the actual installed OpenTUI 0.3.2 and inspected the render buffer's hyperlink metadata directly. OpenTUI only attaches native OSC 8 hyperlink metadata to the **URL substring** of a `[label](url)` Markdown link (e.g. the `https://kilo.ai` text shown in `Kilo (https://kilo.ai)` with the default `conceal` setting) — never to the label word, and never to a bare URL with no Markdown link syntax, in any conceal mode.

Given this, `AssistantLink.url()`:
1. Tries OpenTUI's native hyperlink metadata first (covers the URL substring of labeled links).
2. Falls back to a read-only scan of the rendered character buffer for bare HTTP(S) URLs, with row-to-row continuation so a bare URL that OpenTUI word-wraps across terminal rows still resolves to its full URL from either visual row.

Net effect: clicking the visible URL text of a labeled link, or clicking anywhere in a bare URL (including a wrapped one), opens the link. Clicking only the styled label word itself (e.g. just "Kilo") does not, because OpenTUI exposes no hyperlink metadata there to resolve — this is a real OpenTUI limitation, not a bug in this change. This adjustment was discussed and accepted before continuing implementation.

## Testing
- `bun test ./test/kilocode/cli/cmd/tui/assistant-link.test.tsx` — 10 pass
- `bun run typecheck` (packages/opencode) — clean
- `bun run script/check-opencode-annotations.ts` — all shared upstream changes annotated
- `git diff --check` — clean
- `bunx changeset status` — one pending patch for `@kilocode/cli`
- Fresh independent code review completed with no findings
- Manual macOS Ghostty click-through E2E is still pending (GUI mouse interaction that this session cannot automate); the requesting user has accepted deferring it and will verify before merge.

## Non-goals (unchanged)
Does not touch explicit dialog/auth/retry/notification links, file links, mail links, custom schemes, the Markdown renderer, OpenTUI itself, or the dependency lockfile. No behavior change for iTerm2, Terminal.app, VS Code, Windows Terminal, Linux terminals, tmux, or SSH sessions.

### CODE REVIEW

**Comment by marius-kilocode** [comment]
@iscekic this is different from the default terminal behavior right? In an xterm instance the link underlines on hover and is  clickable via cmd click. Would that not be possible on ghostty as well?

image

**Comment by iscekic** [comment]
Fixed the Kilobot finding: the bare-URL wrap-continuation logic hardcoded buffer column 0 as the next/previous row's start, but the real `assistant-text-` box renders with `paddingLeft={3}` nested inside further ancestor padding, so wrapped content never actually starts at column 0 in production. It now dynamically finds each row's own leftmost/rightmost non-blank content column instead of assuming a fixed offset, on both the leftward and rightward continuation paths (the leftward path had the same class of bug, just not yet triggered by a test case).

Updated the test fixture to render inside the same nested-padding structure as production (`paddingLeft={2}` ancestor + `paddingLeft={3}` on the assistant-text box) so the wrap tests actually exercise non-zero column offsets and would have caught this.

Also found and fixed a related edge case while stress-testing: the bare-URL character-class includes `(` and `)` (they're valid RFC 3986 sub-delimiters), so a URL wrapped in prose like `(see https://kilo.ai)` scanned in with the trailing `)` attached. Added trailing-punctuation trimming (sentence punctuation unconditionally, unbalanced closing brackets/parens/braces) while preserving a genuinely balanced parenthesis that's part of the URL itself (e.g. `https://en.wikipedia.org/wiki/Foo_(bar)`), with tests for both cases.

@marius-kilocode re: your question — you're right that this differs from xterm's default OSC 8 Cmd-click behavior, but only in one specific terminal. Ghostty forwards Cmd-click to the app as an indistinguishable plain left click under SGR mouse reporting (there's no way for the app to tell them apart there — see the linked Ghostty discussion in the PR description's context), so OpenTUI's normal Cmd-click-vs-plain-click distinction that xterm supports isn't available in Ghostty specifically. This change only activates in macOS Ghostty; every other terminal (including xterm) keeps the existing native OSC 8 behavior and its own Cmd-click/Ctrl-click gesture, completely untouched.

**Comment by marius-kilocode** [comment]
@iscekic if we can't do the cmd click in this scenario, can we do the underlining on hover?

**Comment by iscekic** [comment]
@marius-kilocode Good question. Two things worth separating:

- The underline-on-hover in your screenshot is the terminal's own native OSC 8 rendering (this PR doesn't touch that at all — it's untouched on every terminal, including Ghostty, since Ghostty does emit/consume OSC 8 for the visual hyperlink styling; it's specifically the *click gesture disambiguation* that's broken there, not the hyperlink rendering itself). So Ghostty should already underline these links on hover today, independent of this change — that's OpenTUI/Ghostty native behavior, not something this PR adds or could meaningfully change.
- What this PR adds is purely the *click handling* fallback: when Ghostty reports a click, treat it as "open the link" for HTTP(S) links inside assistant text, because Ghostty can't tell the app whether Cmd was held.

If the hover underline isn't actually showing in Ghostty for you, that would be a separate, pre-existing OpenTUI/Ghostty OSC 8 rendering gap unrelated to click handling, and worth its own investigation rather than folding into this fix's scope. Happy to file that separately if you're seeing no underline at all currently.

File: packages/opencode/src/kilocode/cli/cmd/tui/assistant-link.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    cy = y
+    for (;;) {
+      if (cx < buffer.width - 1) {
+        const ch = charAt(buffer, cx + 1, cy)
+        if (URL_CHAR.test(ch)) {
+          text += ch
+          cx++
+          continue
+        }
+        if (blankToRowEnd(buffer, cy, cx + 1) && cy < buffer.height - 1 && URL_CHAR.test(charAt(buffer, 0, cy + 1))) {
```
[the remark this answers is not in the document]
Comment (iscekic) [reply packages/opencode/src/kilocode/cli/cmd/tui/assistant-link.ts]: (bot) Fixed in fd114cac50: wrapped bare URLs now derive each row’s first/last non-blank content column instead of assuming column 0, and the regression fixture mirrors the production nested left padding. The targeted wrapped-URL test passes on the latest head.

**Comment by iscekic** [comment]
Well, I can't seem to get the agent to produce something that works across the board, and I can't spare the time to dive deeper. Closing, might follow up in the future.
